### PR TITLE
feat: database create event endpoint

### PR DESCRIPTION
**Pull request by erickexinlu**
## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,6 +5,8 @@ FRONTEND_DIR=./frontend
 include .env
 export $(shell sed 's/=.*//' .env)
 
+br_backend: build_backend run
```
Comment (lou-weng) [inline Makefile:8]: wats this for

Comment (erickexinlu) [reply Makefile:8]: i just made so i wouldn't have to run make build_backend and then make run when i make some edit to the backend, so i don't have to build the frontend every time and can just use one command

File: backend/src/services/data/event_data_service.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	fmt.Println("\nRow inserted successfully!")
+}
+
+func UpdateEvent(env *models.Env, event models.Event) {
+	query := `UPDATE Events
+			  SET EventName = ($2),
+			  	  StartTime = ($3),
+				  EndTime = ($4)
+			  WHERE EventID = ($1)`
```
Comment (lou-weng) [inline backend/src/services/data/event_data_service.go:32]: Screenshot 2023-01-07 at 2 13 58 AM

File: backend/src/services/data/event_data_service.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import (
+	"context"
+	"fmt"
+	"log"
+
+	models "schedulii/src/models"
+)
+
+// Helper function to add a event to the database
+func CreateEvent(env *models.Env, event models.Event) {
```
Comment (lou-weng) [inline backend/src/services/data/event_data_service.go:12]: return an error type

if no error, return nil

**Review by lou-weng — CHANGES_REQUESTED** [review]

File: backend/go.mod [hunk 4]
```diff
@@ -11,6 +11,7 @@ require (
 )
 
 require (
+	github.com/georgysavva/scany/v2 v2.0.0 // indirect
```
Comment (lou-weng) [inline backend/go.mod:14]: remove

File: backend/src/main.go [hunk 5]
```diff
@@ -10,6 +10,7 @@ import (
 	models "schedulii/src/models"
 	router "schedulii/src/routes"
 
+	// remove this later
```
Comment (lou-weng) [inline backend/src/main.go:13]: what is this comment for

File: backend/src/services/data/event_data_service.go (hunk 2 above)
Comment (erickexinlu) [reply backend/src/services/data/event_data_service.go:32]: image
argh it's being weird again, looks normal on my vscode

File: backend/src/services/data/event_data_service.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		context.Background(),
+		query,
+		event.EventId,
+		event.EventName,
+		event.StartTime,
+		event.EndTime)
+	if err != nil {
+		log.Fatalf("Unable to update value: %v", err)
+	}
+	fmt.Println("\nRow updated successfully!")
```
Comment (lou-weng) [inline backend/src/services/data/event_data_service.go:43]: return error type for function, nil if there is no error

File: backend/src/services/data/event_data_service.go [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+func ReadEvent(env *models.Env, event models.Event) models.Event {
```
Comment (lou-weng) [inline backend/src/services/data/event_data_service.go:46]: function should return type of (models.Events, error)

**Review by lou-weng — CHANGES_REQUESTED** [review]

File: backend/src/services/data/event_data_service.go (hunk 2 above)
Comment (erickexinlu) [reply backend/src/services/data/event_data_service.go:32]: to be safe i made the entire line out of spaces for the next commit

File: backend/src/main.go (hunk 5 above)
Comment (erickexinlu) [reply backend/src/main.go:13]: removed

File: Makefile (hunk 1 above)
Comment (lou-weng) [reply Makefile:8]: I would change the name of the command br_backend is confusing

**Review by lou-weng — APPROVED** [review]

**Pull request merged**